bmorph/core/bmorph.py

```diff
@@ -132,25 +202,49 @@ def bmorph(raw_ts, raw_cdf_window, raw_bmorph_window,
     assert isinstance(train_ts, pd.Series)
     assert isinstance(training_window, slice)
 
-    # Create the CDFs that are used for morphing the raw_ts. The mapping is
-    # based on the training_window
-    truth_cdf = truth_ts[training_window].rolling(
-        window=nsmooth, min_periods=1, center=True).mean()
-    train_cdf = train_ts[training_window].rolling(
-        window=nsmooth, min_periods=1, center=True).mean()
-
     # Smooth the raw Series
     raw_smoothed_ts = raw_ts.rolling(
         window=nsmooth, min_periods=1, center=True).mean()
-
-    # Calculate the bmorph multipliers based on the smoothed time series and
-    # PDFs
-    bmorph_multipliers = edcdfm(raw_smoothed_ts[raw_bmorph_window],
-                                raw_smoothed_ts[raw_cdf_window],
-                                train_cdf, truth_cdf)
-
-    # Apply the bmorph multipliers to the raw time series
-    bmorph_ts = bmorph_multipliers * raw_ts[raw_bmorph_window]
+    
+    # Check if using edcdfm or mdcdedcdfm through second variable being added  
+    # for the raw and train series because truth can be set as train later
+    if (raw_y is None) or (truth_y is None) or (train_y is None):
+        # Create the CDFs that are used for morphing the raw_ts. The mapping is
+        # based on the training_window
+        truth_cdf = truth_ts[training_window].rolling(
+            window=nsmooth, min_periods=1, center=True).mean()
+        train_cdf = train_ts[training_window].rolling(
+            window=nsmooth, min_periods=1, center=True).mean()
+
+        # Calculate the bmorph multipliers based on the smoothed time series and
+        # PDFs
+        bmorph_multipliers = edcdfm(raw_smoothed_ts[raw_bmorph_window],
+                                    raw_smoothed_ts[raw_cdf_window],
+                                    train_cdf, truth_cdf)
+
+        # Apply the bmorph multipliers to the raw time series
+        bmorph_ts = bmorph_multipliers * raw_ts[raw_bmorph_window]
+        
+    else:
+        # Continue Type Checking additionally series
+        assert isinstance(raw_y, pd.Series)
+        assert isinstance(truth_y, pd.Series)
+        assert isinstance(train_y, pd.Series)
+        
+        # smooth the raw y series
+        raw_smoothed_y = raw_y.rolling(
+            window=nsmooth, min_periods=1, center=True).mean()
```

Comment:
I think you're missing the rest of the smoothing here. We need to also be smoothing `truth_ts` and `train_ts` just as in the other half of the conditional. We should also be smoothing the other `y` variables for consistency.

Reply:
oh, I see now, thanks

Reply:
should the truth and the train be smoothed over the whole series or just the `training_window`?